ppl/archive/index/proc.go

```diff
@@ -0,0 +1,167 @@
+package index
```

Comment:
This code was copied from archive/proc.go. I'll remove archive/proc.go in a subsequent pr.